mcs/class/System/System.Diagnostics/PerformanceCounter.cs

```diff
@@ -146,7 +159,22 @@ void UpdateInfo ()
 			// need to free the previous info
 			if (impl != IntPtr.Zero)
 				Close ();
-			impl = GetImpl (categoryName, counterName, instanceName, machineName, out type, out is_custom);
+
+			impl = IntPtr.Zero;
+
+			if (ValidMachine (machineName)) {
+				unsafe {
+					fixed (char* fixed_categoryName = (categoryName != null) ? categoryName : null,
```

Comment:
the null check is redundant `fixed (char* fixed_categoryName = categoryName)` will give you same result